diff --git a/node_modules/react-native-paper/lib/commonjs/components/Dialog/DialogActions.js b/node_modules/react-native-paper/lib/commonjs/components/Dialog/DialogActions.js
index 2e58d97..457f76a 100644
--- a/node_modules/react-native-paper/lib/commonjs/components/Dialog/DialogActions.js
+++ b/node_modules/react-native-paper/lib/commonjs/components/Dialog/DialogActions.js
@@ -46,7 +46,7 @@ const DialogActions = props => {
   return /*#__PURE__*/React.createElement(_reactNative.View, _extends({}, props, {
     style: [isV3 ? styles.v3Container : styles.container, props.style]
   }), React.Children.map(props.children, (child, i) => /*#__PURE__*/React.isValidElement(child) ? /*#__PURE__*/React.cloneElement(child, {
-    compact: true,
+    compact: props.compact ?? false,
     uppercase: !isV3,
     style: [isV3 && {
       marginRight: i + 1 === actionsLength ? 0 : 8
diff --git a/node_modules/react-native-paper/lib/commonjs/components/List/ListItem.js b/node_modules/react-native-paper/lib/commonjs/components/List/ListItem.js
index 2aa6cb1..f011c8d 100644
--- a/node_modules/react-native-paper/lib/commonjs/components/List/ListItem.js
+++ b/node_modules/react-native-paper/lib/commonjs/components/List/ListItem.js
@@ -84,7 +84,11 @@ const ListItem = _ref => {
     }, description);
   };
   const renderTitle = () => {
-    const titleColor = theme.isV3 ? theme.colors.onSurface : (0, _color.default)(theme.colors.text).alpha(0.87).rgb().string();
+    const titleColor = theme.isV3
+      ? !rest.disabled
+        ? theme.colors.onSurface
+        : theme.colors.onSurfaceDisabled
+      : (0, _color.default)(theme.colors.text).alpha(0.87).rgb().string();
     return typeof title === 'function' ? title({
       selectable: false,
       ellipsizeMode: titleEllipsizeMode,
@@ -100,7 +104,11 @@ const ListItem = _ref => {
       maxFontSizeMultiplier: titleMaxFontSizeMultiplier
     }, title);
   };
-  const descriptionColor = theme.isV3 ? theme.colors.onSurfaceVariant : (0, _color.default)(theme.colors.text).alpha(0.54).rgb().string();
+  const descriptionColor = theme.isV3
+     ? !rest.disabled 
+        ? theme.colors.onSurfaceVariant
+        : theme.colors.onSurfaceDisabled
+    : (0, _color.default)(theme.colors.text).alpha(0.54).rgb().string();
   return /*#__PURE__*/React.createElement(_TouchableRipple.default, _extends({}, rest, {
     style: [theme.isV3 ? styles.containerV3 : styles.container, style],
     onPress: onPress,
diff --git a/node_modules/react-native-paper/lib/module/components/Dialog/DialogActions.js b/node_modules/react-native-paper/lib/module/components/Dialog/DialogActions.js
index dc315bb..ae4f06d 100644
--- a/node_modules/react-native-paper/lib/module/components/Dialog/DialogActions.js
+++ b/node_modules/react-native-paper/lib/module/components/Dialog/DialogActions.js
@@ -38,7 +38,7 @@ const DialogActions = props => {
   return /*#__PURE__*/React.createElement(View, _extends({}, props, {
     style: [isV3 ? styles.v3Container : styles.container, props.style]
   }), React.Children.map(props.children, (child, i) => /*#__PURE__*/React.isValidElement(child) ? /*#__PURE__*/React.cloneElement(child, {
-    compact: true,
+    compact: props.compact ?? false,
     uppercase: !isV3,
     style: [isV3 && {
       marginRight: i + 1 === actionsLength ? 0 : 8
diff --git a/node_modules/react-native-paper/lib/module/components/List/ListItem.js b/node_modules/react-native-paper/lib/module/components/List/ListItem.js
index fcbf760..2a9beda 100644
--- a/node_modules/react-native-paper/lib/module/components/List/ListItem.js
+++ b/node_modules/react-native-paper/lib/module/components/List/ListItem.js
@@ -75,7 +75,11 @@ const ListItem = _ref => {
     }, description);
   };
   const renderTitle = () => {
-    const titleColor = theme.isV3 ? theme.colors.onSurface : color(theme.colors.text).alpha(0.87).rgb().string();
+    const titleColor = theme.isV3
+      ? !rest.disabled
+        ? theme.colors.onSurface
+        : theme.colors.onSurfaceDisabled
+      : (0, _color.default)(theme.colors.text).alpha(0.87).rgb().string();
     return typeof title === 'function' ? title({
       selectable: false,
       ellipsizeMode: titleEllipsizeMode,
@@ -91,7 +95,11 @@ const ListItem = _ref => {
       maxFontSizeMultiplier: titleMaxFontSizeMultiplier
     }, title);
   };
-  const descriptionColor = theme.isV3 ? theme.colors.onSurfaceVariant : color(theme.colors.text).alpha(0.54).rgb().string();
+  const descriptionColor = theme.isV3
+     ? !rest.disabled 
+        ? theme.colors.onSurfaceVariant
+        : theme.colors.onSurfaceDisabled
+    : (0, _color.default)(theme.colors.text).alpha(0.54).rgb().string();
   return /*#__PURE__*/React.createElement(TouchableRipple, _extends({}, rest, {
     style: [theme.isV3 ? styles.containerV3 : styles.container, style],
     onPress: onPress,
diff --git a/node_modules/react-native-paper/lib/typescript/components/Dialog/DialogActions.d.ts b/node_modules/react-native-paper/lib/typescript/components/Dialog/DialogActions.d.ts
index d349a10..1ecfeeb 100644
--- a/node_modules/react-native-paper/lib/typescript/components/Dialog/DialogActions.d.ts
+++ b/node_modules/react-native-paper/lib/typescript/components/Dialog/DialogActions.d.ts
@@ -6,6 +6,8 @@ export declare type Props = React.ComponentPropsWithRef<typeof View> & {
      * Content of the `DialogActions`.
      */
     children: React.ReactNode;
+    /** Whether Dialog Action buttons should be compact */
+    compact?: boolean;
     style?: StyleProp<ViewStyle>;
     /**
      * @optional
diff --git a/node_modules/react-native-paper/src/components/Dialog/DialogActions.tsx b/node_modules/react-native-paper/src/components/Dialog/DialogActions.tsx
index b8f7b0b..bf1cda9 100644
--- a/node_modules/react-native-paper/src/components/Dialog/DialogActions.tsx
+++ b/node_modules/react-native-paper/src/components/Dialog/DialogActions.tsx
@@ -10,6 +10,8 @@ export type Props = React.ComponentPropsWithRef<typeof View> & {
    * Content of the `DialogActions`.
    */
   children: React.ReactNode;
+  /** Whether Dialog Action buttons should be compact */
+  compact?: boolean;
   style?: StyleProp<ViewStyle>;
   /**
    * @optional
@@ -57,7 +59,8 @@ const DialogActions = (props: Props) => {
       {React.Children.map(props.children, (child, i) =>
         React.isValidElement(child)
           ? React.cloneElement(child as React.ReactElement<any>, {
-              compact: true,
+              // Allow overriding compact prop via 'DialogAction'
+              compact: props.compact ?? false,
               uppercase: !isV3,
               style: [
                 isV3 && {
diff --git a/node_modules/react-native-paper/src/components/List/ListItem.tsx b/node_modules/react-native-paper/src/components/List/ListItem.tsx
index ff82dbf..2a6b256 100644
--- a/node_modules/react-native-paper/src/components/List/ListItem.tsx
+++ b/node_modules/react-native-paper/src/components/List/ListItem.tsx
@@ -188,7 +188,9 @@ const ListItem = ({
 
   const renderTitle = () => {
     const titleColor = theme.isV3
-      ? theme.colors.onSurface
+      ? !rest.disabled
+        ? theme.colors.onSurface
+        : theme.colors.onSurfaceDisabled
       : color(theme.colors.text).alpha(0.87).rgb().string();
 
     return typeof title === 'function' ? (
@@ -212,7 +214,9 @@ const ListItem = ({
   };
 
   const descriptionColor = theme.isV3
-    ? theme.colors.onSurfaceVariant
+    ? !rest.disabled
+      ? theme.colors.onSurfaceVariant
+      : theme.colors.onSurfaceDisabled
     : color(theme.colors.text).alpha(0.54).rgb().string();
 
   return (
